subprojects/core/src/main/java/org/gradle/api/internal/CompositeDomainObjectSet.java

```diff
@@ -163,4 +145,135 @@ public void all(Action<? super T> action) {
         }
     }
 
+    private final static class DomainObjectCompositeCollection<T> implements Collection<T>, WithEstimatedSize {
+
+        private final List<DomainObjectCollection<? extends T>> store = Lists.newLinkedList();
```

Comment:
It's always created: it's used in the constructor, which will always add at least one element to it.